server/src/main/java/com/cloud/api/ApiResponseHelper.java

```diff
@@ -2583,9 +2583,13 @@ public NetworkResponse createNetworkResponse(ResponseView view, Network network)
                 Domain domain = ApiDBUtils.findDomainById(domainNetworkDetails.first());
                 if (domain != null) {
                     response.setDomainId(domain.getUuid());
-
-                    StringBuilder domainPath = new StringBuilder("ROOT");
-                    (domainPath.append(domain.getPath())).deleteCharAt(domainPath.length() - 1);
+                    StringBuilder domainPath = new StringBuilder();
+                    if(ObjectUtils.isEmpty(domain.getPath()) || domain.getPath().equals("/")){
```

Comment:
not sure what you intent to do here @isupersky . it seems now that you append / to / resulting in //???
also you keep changing this in three locations. Would be good to have a method so we can change the code as needed in only one location.

Reply:
@DaanHoogland there are two variables here, domain.getPath() and domainPath.
domainPath is initially empty `StringBuilder domainPath = new StringBuilder();`  , hence it will only be initialised with single "/".

Thanks for the second suggestion,  I'll move the code repetition into a method.